lib/metasploit/framework/login_scanner/zabbix.rb

```diff
@@ -48,7 +48,7 @@ def check_setup
               return "Unexpected HTTP response code #{res.code} (is this really Zabbix?)"
             end
 
-            if res.body.to_s !~ /Zabbix ([^\s]+) Copyright .* by Zabbix/m
+            if res.body.to_s !~ /Zabbix/
```

Comment:
This will match on anything that contains `Zabbix` in its page. I could make a page that contains a reference to Zabbix in passing, such as in a comment on a blog, and this module will think that the target is a Zabbix target. This is really not what you want to be doing.

If you are trying to change this to support multiple versions then what I suggest you do is to keep the current regex, and then add a new check specifically for the new string you are seeing in the application. This will make sure that you still keep support for the old versions of the app whilst also adding in support for the new version string you are seeing. Again try make it as specific as possible, as otherwise you run a high change of false positives which we want to avoid.

Reply:
I tested with versions 3,4, and 5 and noticed that the Copyright line is no longer part of the return code after successful login.
I did notice that the Author line is consistent, as far back as version 2.0.21 of Zabbix.  Suggesting that the regex change to the following:
            if res.body.to_s !~ /name="Author" content="Zabbix SIA"/m
This would tighten up the regex as well as support old and new versions.

Reply:
Hmm I'll take a look at this tomorrow but the issue with just changing this regex, whilst I do think it is a good idea, is that the previous regex included a group which is then assigned to `self.version` on line 55 with the assignment `self.version = $1` (see https://github.com/rapid7/metasploit-framework/pull/14297/files#diff-9e2c7de768883fab840603662ce0e5aeecf96279c32647fd334bd3e847226edeR55 )

I'll need to examine this further to confirm but there is a the potential that this may be an issue if this `self.version` value is actually being used within the code.

My initial inspection makes me think this `self.version` line is actually used in the output as indicated by  https://github.com/rapid7/metasploit-framework/pull/14297/files#diff-8fe89a467adae71b015090ebd7c121feffad8826cd9a94d734e32349fa45d881R41 not showing any version number in the output like it would normally, most likely as a result of your changes to the regex.

Whilst in this case it doesn't strictly speaking break anything critical it does prevent what appears to be an intentional version enumeration part of the code here.

Reply:
Okay after further review I think the best regex might be something like this:

`<a target="_blank" class="grey link-alt" href="http[sS]{0,1}:\/\/www\.zabbix\.com\/documentation\/(\d+\.\d+)\/">Help<\/a>`

This seems to match both on the latest version of Zabbix, 5.0.5 LTS, as well as version 4.4, 4.0, and 3.2 of Zabbix, and also ensures that we are still grabbing the version information so that it can be output by the scanner.

We would then keep the other regex that is in the code already and run this second regex if the one in the code already fails, such that it would basically go "if the first regex fails, try this second regex. If both fail, then the target likely isn't a Zabbix machine".

Reply:
Applied the regex fixes in https://github.com/rapid7/metasploit-framework/pull/14297/commits/bd578324941b2ef1f65b452c80c45875f9bf9b68 and nested them so that we still have support for the old version. New output now includes the version info as well as a result of updating this regex.

```
msf6 auxiliary(scanner/http/zabbix_login) > run
[*] 172.29.121.85:80 - Found Zabbix version 5.0
[*] 172.29.121.85:80 - This Zabbix instance has disabled Guest mode
[+] 172.29.121.85:80 - Success: 'Admin:zabbix'
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/http/zabbix_login) >
```